lib/urbanairship/http_client.rb

```diff
@@ -0,0 +1,89 @@
+module Urbanairship
+  class HttpClient
+
+    # Make an HTTP Request
+    def self.request(*args)
+      HttpClient::Request.new(*args).send_request
+    end
+
+    class Response
+      attr_reader :body, :code, :headers
+
+      def initialize(http_response)
+        @body    = parse_json(http_response.body)
+        @code    = http_response.code.to_i
+        @headers = http_response.each_header.to_a
+      end
+
+      def [](attribute)
+        send(attribute) if respond_to?(attribute)
+      end
+
+      private
+
+      def parse_json(body)
+        JSON.parse(body.to_s)
+      rescue JSON::ParserError
+        {}
+      end
```

Comment:
Rescuing JSON::ParserError here. Should it return `nil` or empty string instead of empty hash?